File: 2021-09/x만큼간격이있는n개의숫자.js
function solution(x, n) {
    const answer = [];
    answer.push(x);
    for(let i = 1, count = x; i < n; i++) {
        answer.push(count += x);
    }
    return answer;
}

function solution2(x, n) {
    const answer = [];
    for(let i = 1; i <= n; i++) {
        answer.push(x * i);
    }
    return answer;
}

console.log(solution(2, 5)); // [2, 4, 6, 8, 10]
console.log(solution(4, 3)); // [4, 8, 12]
console.log(solution(-4, 2)); // [-4, -8]

File: programmers/src/main/java/level1/x만큼간격이있는n개의숫자.java
package level1;

public class x만큼간격이있는n개의숫자 {

	public long[] solution(int x, int n) {
		long[] answer = new long[n];

		answer[0] = x;
		long nowSum = x;

		for (int i = 1; i < n; i++) {
			nowSum += x;
			answer[i] = nowSum;
		}
		return answer;
	}

	/**
	 * 배열 이전껄 보고 더해주는게 멋있어보였음
	 */
	public static long[] solution2(int x, int n) {
		long[] answer = new long[n];
		answer[0] = x;

		for (int i = 1; i < n; i++) {
			answer[i] = answer[i - 1] + x;
		}

		return answer;
	}

}
